### PR TITLE
[CI] Install dill in ci

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116215
* __->__ #116214
* #116230

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/test_dataloader.py [hunk 1]
```diff
@@ -1613,6 +1599,18 @@ def test_multiprocessing_iterdatapipe(self):
                                   for t in
                                   self._get_data_loader(datapipe, multiprocessing_context=ctx, **dl_common_args)])
 
+    @skipIfNoNumpy
+    @unittest.skipIf(IS_JETSON, "Not working on Jetson")
+    def test_multiprocessing_iterdatapipe(self):
+        self._test_multiprocessing_iterdatapipe(with_dill=False)
+
+    @unittest.expectedFailure
```
Comment (peterbell10) [inline test/test_dataloader.py:1607]: This test fails in CI when the `HAS_DILL` code is active, so keeping the non-dill version active and xfailing the version with dill for now.

**Comment by lezcano** [comment]
ping @malfet

File: .ci/docker/requirements-ci.txt [hunk 2]
```diff
@@ -25,6 +25,11 @@ coremltools==5.0b5
 #Pinned versions:
 #test that import:
 
+dill
```
Comment (malfet) [inline .ci/docker/requirements-ci.txt:28]: Please specify the version that you want to use.

**Review by malfet — APPROVED** [review]
Looks good to me, but it feels like this PR solves the problem when tests are running with dill, but does not test that PyTorch will not accidentally introduce unconditional dependency on dill, because its present in all CI configs.

Can we somehow keep one of the Linux CPU configs without dill(i.e. modify one of the test configs not uninstall dill before running the tests)

**Pull request closed without merging**